### PR TITLE
ASP support member lookup

### PR DESCRIPTION
**Pull request by worksofliam**
### Changes

Similar to opening files, `memberResolve` will now look in the user defined ASP to find a member. Additionally, the API can accept an ASP and will use that instead.

Also solves a bug with the `qualifyPath` which wasn't handling `QSYS` correctly.

### Checklist

* [x] have tested my change
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] eslint is not complaining
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/codefori/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [ ] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

File: src/api/IBMiContent.ts [hunk 1]
```diff
@@ -678,7 +678,15 @@ export default class IBMiContent {
...
-      .map(file => `/QSYS.LIB/${file.library}.LIB/${file.name}.FILE/${member}.MBR`)
+      .map(file => {
+        const asp = file.asp || this.config.sourceASP;
+        if (asp && asp.length > 0) {
+          return [`/${asp}/QSYS.LIB/${file.library}.LIB/${file.name}.FILE/${member}.MBR`, `/QSYS.LIB/${file.library}.LIB/${file.name}.FILE/${member}.MBR`].join(` `);
+        } else {
+          return `/QSYS.LIB/${file.library}.LIB/${file.name}.FILE/${member}.MBR`;
+        }
+      })
+      .map(file => Tools.escapePath(file))
```
Comment (worksofliam) [inline src/api/IBMiContent.ts:689]: @chrjorgensen Is this the valid way to use escape path? The tests continue to pass and this should ensure that `$` can be used in the path itself.

Comment (chrjorgensen) [reply src/api/IBMiContent.ts:689]: @worksofliam LGTM - even though I haven't had anything to do with `Tools.escapePath`... you may be thinking of `Tools.sanitizeLibraryNames`?

Using `Tools.escapePath` should be okay, when the library is not preceded by a space (which caused trouble in the shell when the library name started with `#`).

Comment (worksofliam) [reply src/api/IBMiContent.ts:689]: Supposed I better use `Tools.sanitizeLibraryNames` too!

Comment (worksofliam) [reply src/api/IBMiContent.ts]: @chrjorgensen I actually used both in a new commit. Made a well needed improved to one of our existing APIs.

**Comment by worksofliam** [comment]
Tests continue to pass.

**Pull request merged**